src/mercury-common/mercury/common/configuration.py

```diff
@@ -79,3 +84,368 @@ def get_configuration(filename):
+        self.configuration = {}
+
+        self.argument_parser.add_argument('-c',
+                                          '--configuration',
+                                          default=self.configuration_file,
+                                          help='Path to the configuration file')
+
+        # Add logging at the common level, every program will need this
+        # advanced logging facilities must be implemented at the program
+        # level
+
+        self.add_option('log_level',
+                        '--log-level',
+                        'MERCURY_LOG_LEVEL',
+                        'logging.level',
+                        default='INFO',
+                        help_string='Log level, INFO | DEBUG | ERROR | WARNING')
+
+        self.add_option('log_format',
+                        '--log-format',
+                        'MERCURY_LOG_FORMAT',
+                        'logging.format',
+                        default=log_format,
+                        help_string='Desired logging format')
+
+    def set_configuration(self):
+        """ Parse the configuration """
+
+        # self.configuration file is the default option
+        if self.argparse_namespace.configuration != self.configuration_file:
+            # Configuration file is explicitly defined on the command line
+            self.configuration = configuration_from_yaml(
+                self.argparse_namespace.configuration)
+
+        # Search for the configuration file in the standard location
+        config_file = find_config(self.configuration_file,
+                                  self.config_search_dirs)
+        if config_file:
+            self.configuration = configuration_from_yaml(config_file)
+
+    @staticmethod
+    def get_by_namespace(dictionary, namespace):
+        """
+        Used to access a dictionary values using dot notation
+
+        For instance:
+
+        .. code-block::
+
+            {'one': {'two': 'three'}}
+
+            >> d['one']['two']
+            'three'
+            >> get_by_namespace('one.two')
+            'three'
+        :param dictionary: The dictionary we are searching
+        :param namespace: Dot notated namespace specifier
+        :type namespace: str
+        :return: The value the namespace resolves to or None
+        """
+
+        # creating copy of reference for readability
+        value = dictionary
+        for k in namespace.split('.'):
+            try:
+                value = value[k]
+            except KeyError:
+                return
+        return value
+
+    @staticmethod
+    def set_by_namespace(dictionary, namespace, value):
+        """
+        Like get_by_namespace, but set a value rather than returning it
+        :param dictionary: The dictionary we are modifying
+        :param namespace:
+        :param value:
+        :return:
+        """
+        expanded = namespace.split('.')
+        for k in expanded[:-1]:
```

Comment:
Q: Why is it dropping the last entry?

Presumably, this would take `my.config.option` and only access `my.config` and drop `option`. Is that b/c the value is the last part of the namespace format? If so, why have a `value` parameter and not just include it in the `namespace`? 

```python
>>> x = "my.config.option"
>>> y = x.split('.')
>>> y
['my', 'config', 'option']
>>> y[:-1]
['my', 'config']
```

Reply:
So the last element of the namespace is the value reference. We need the container of the value reference so that we can update or set that value:
```
In [1]: d = {'my': {'nested': {'value': 1}}}

In [2]: d
Out[2]: {'my': {'nested': {'value': 1}}}

In [4]: value_ref = d['my']['nested']['value']

In [5]: value_ref
Out[5]: 1

In [6]: dict_ref = d['my']['nested']

In [7]: dict_ref
Out[7]: {'value': 1}

In [8]: dict_ref['value'] = 'new_value'

In [9]: d
Out[9]: {'my': {'nested': {'value': 'new_value'}}}
```

Reply:
👍 for documentation , I'll add it. the use of arbitrary indexes is a bit confusing.. in fact, I didn't add that in until I tested the code and found it wasn't working as expected